docs/reference/content/bson/macros.md

```diff
@@ -0,0 +1,86 @@
++++
+date = "2016-11-20T14:14:00-00:00"
+title = "Macros"
+[menu.main]
+  parent = "BSON"
+  weight = 30
+  pre = "<i class='fa fa-cog'></i>"
++++
+
+## Macros
+
+New in 2.0 the scala driver allows you to use case classes to represent documents in a collection via the 
+[`Macros`]({{< apiref "org.mongodb.scala.bson.codecs.Macros" >}}) helper.  Simple case classes and nested case classes are supported. 
+Hierarchical modelling can be achieve by using a sealed trait and having case classes implement the parent trait.
+
+Many simple scala types are supported and they will be marshaled into their corresponding 
+[`BsonValue`]({{< apiref "org.mongodb.scala.bson.BsonValue" >}}) type. Below is a list of Scala types and their type safe Bson representation:
+   
+| Scala type                        | BSON type         |
+|-----------------------------------|-------------------|
+| case class                        | Document          |
+| `Iterable`                        | Array             |
+| `Date`                            | Date              |
+| `Boolean`                         | Boolean           |
+| `Double`                          | Double            |
```

Comment:
We currently don't support type massaging and don't plan to add support for this release.  Once the java driver supports it we'll get it for free :)

Is there a Jira ticket to track it?